pydantic/types.py

```diff
@@ -431,14 +431,10 @@ def validate(cls, value: Union[str]) -> Union[str]:
 
         if cls.regex:
             if not re.match(cls.regex, value):
-                raise errors.StrRegexError(pattern=cls._get_pattern(cls.regex))
+                raise errors.StrRegexError(pattern=get_pattern(cls.regex))
 
         return value
 
-    @staticmethod
-    def _get_pattern(regex: Union[str, Pattern[str]]) -> str:
```

Comment:
We can't remove this function in case people are using it.

People could also have overridden the method to customise behaviour. 

Better to keep this and use it as we were before.